File: Gevorgyan.TodoListApp.Application/Commands/Users/Models/AuthenticateUserInputModel.cs
using System.ComponentModel.DataAnnotations;

namespace Gevorgyan.TodoListApp.Application.Commands.Users.Models
{
    /// <summary>
    /// Input model of data for user authentication
    /// </summary>
    public class AuthenticateUserInputModel : InputModelBase
    {
        /// <summary>
        /// Email
        /// </summary>
        [RequiredInModel]
        [Display(Name = "Email of user")]
        public string UserName { get; set; }

        /// <summary>
        /// Password
        /// </summary>
        [RequiredInModel]
        [Display(Name = "Password of user")]
        [NotTrimable]
        public string Password { get; set; }
    }
}

File: Gevorgyan.TodoListApp.RestApi/ClientApp/src/app/core/models/AuthenticateUserInputModel.ts



// for AUTO-GENERATION!



/*
 * Input model of data for user authentication
 */
export interface IAuthenticateUserInputModel {
    
    /*
     * Email
     */
    userName: string;

    /*
     * Password
     */
    password: string;
}
    
/*
 * Input model of data for user authentication
 */
export class AuthenticateUserInputModel implements IAuthenticateUserInputModel {
    
    /*
     * Email
     */
    userName: string;
    
    /*
     * Password
     */
    password: string;
    
    constructor(data?: IAuthenticateUserInputModel) {
        
        if (data) {
            for (var property in data) {
                if (data.hasOwnProperty(property))
                    (<any>this)[property] = (<any>data)[property];
            }
        }
    }

    init(data?: any) {
        if (data) {
            
            this.userName = data["userName"];
            this.password = data["password"]; 
        }
    }

    static fromJS (data: any): AuthenticateUserInputModel {
        data = typeof data === 'object' ? data : {};
        let result = new AuthenticateUserInputModel;
        result.init(data);
        return result;
    }

    toJSON(data?: any) {
        data = typeof data === 'object' ? data : {};
        
        data["userName"] = this.userName;
        data["password"] = this.password;
        
        return data; 
    }
}    
